File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        stack = []
        for c in s:
            if c=='(' or c=='{' or c=='[' or not stack:
                stack.append(c)
            elif c==')' and stack[-1]=='(':
                stack.pop()
            elif c=='}' and stack[-1]=='{':
                stack.pop()
            elif c==']' and stack[-1]=='[':
                stack.pop()
            else:
                return False
        return len(stack)==0
        """
        :type s: str
        :rtype: bool
        """

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for(char c : s.toCharArray()){
            if(c == '(' || c == '{' || c == '[' || stack.isEmpty())
                stack.push(c);
            else if(!stack.isEmpty()){
                if(stack.peek() == '(' && c == ')')
                    stack.pop();
                else if(stack.peek() == '{' && c == '}')
                    stack.pop();
                else if(stack.peek() == '[' && c == ']')
                    stack.pop();
                else
                    return false;
            }
        }
        return stack.size() == 0;
    }
}